solang-parser/src/diagnostics.rs

```diff
@@ -20,23 +20,24 @@ impl Level {
     }
 }
 
-#[derive(Debug, Eq, Hash, PartialOrd, Ord, PartialEq)]
+#[derive(Clone, Debug, Eq, Hash, PartialOrd, Ord, PartialEq)]
 pub enum ErrorType {
```

Comment:
See above.

Also, I wanted to implement `Clone` for Diagnostic. I don't want to implement `Copy` for diagnostics because I don't want to implicitly copy the structure by accident.